src/regression.js

```diff
@@ -30,12 +30,26 @@
   /**
    * Determine the coefficient of determination (r^2) of a fit from the observations and predictions.
    *
+   * @param {Array<Array<number>>} data - Pairs of observed x-y values, including nulls
+   * @param {Array<Array<number>>} results - Pairs of observed predicted x-y values, including null predictions
+   *
    * @param {Array<Array<number>>} observations - Pairs of observed x-y values
    * @param {Array<Array<number>>} predictions - Pairs of observed predicted x-y values
    *
    * @return {number} - The r^2 value, or NaN if one cannot be calculated.
    */
-  function determinationCoefficient(observations, predictions) {
+  function determinationCoefficient(data, results) {
+    var observations = [];
+    var predictions = [];
+
+    // Remove predictive points for accurate fit
+    data.forEach( function (d, i) {
```

Comment:
> Make the entire library filter out null

I agree that the entire library should be changed to reflect this change, I just haven't had the time to look at the other regressions to even see if they also exhibit the same behavior.

> Retain the original behavior and add a flag for conditionally filtering null

Similar to the above, this option would be worth exploring.

> Don't change the library at all and filter null at the input

I don't think this is an option _if_ you ever want to be able to be able to use this library to fill in missing data, which is pretty pivotal to our use case.

> Why not just create a helper method that does the null filtering then passes it to the regression library?

If you filter out the nulls, then you are no longer filling in the blanks/forecasting.

From the docs for this library: 
> Filling the blanks and forecasting
var data = [[0,1], [32, null] .... [12, 79]];
In any regression, if you use a null value for data, regression-js will fill it using the trend.

Except that it _doesn't_ do the above. It 0 fills. So either the docs are wrong, _or_ we need to make some tweaks to any regressions that don't work. Judging by the issues this might be the only regression that _is_ broken.